takari-lifecycle-plugin/pom.xml

```diff
@@ -120,7 +120,7 @@
     <dependency>
       <groupId>org.eclipse.tycho</groupId>
       <artifactId>org.eclipse.jdt.core</artifactId>
-      <version>3.12.0.v20160516-2131</version>
+      <version>3.12.2.v20161117-1814</version>
```

Comment:
you need to update both org.eclipse.jdt.core and org.eclipse.jdt.compiler.apt to use matching versions from the same eclipse release. ideally, update org.eclipse.osgi too (just to be consistent) but that is not a requirement.